### PR TITLE
Update ATRT subtype

### PR DESCRIPTION
**Pull request by zzgeng**
### Purpose/implementation Section
UPDATED Analysis: ATRT subtyping
#### What scientific question is your analysis addressing?
Assign ATRT tumors to one of 3 subtypes:

ATRT, MYC
ATRT, SHH
ATRT, TYR
using methylation results

#### What was your approach?
archive current scripts 00, 01, and 02 - create an archive folder and move them there - we might want to use some of this information later for classification in the absence of methylation, but for now, we will rely only on methylation results
Create a subtyping script to subtype all PBTA and/or DGD tumor biospecimens which have matched methylation (matched on sample_id). If methylation does not exist for any ATRT samples, those biospecimens can be deemed "ATRT, To be clasified" subtype.
The above subtypes should be listed in the methylation results (similar to IHG subtyping). Use only those with scores >=0.8.
The expected table will have matched tumor DNA/RNA/methylation biospecimen IDS for all ATRTs with new subtypes or a To be classified subtype.

#### What GitHub issue does your pull request address?
https://github.com/PediatricOpenTargets/ticket-tracker/issues/496

#### Which areas should receive a particularly close look?
OpenPedCan-analysis/analyses/molecular-subtyping-ATRT/ATRT_subtyping.R

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

### Results
OpenPedCan-analysis/analyses/molecular-subtyping-ATRT/results/ATRT_subtype_methylation.tsv
#### What types of results are included (e.g., table, figure)?
Table

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jharenza** [comment]
@zzgeng it looks like you have some files pulled over from your last set of PRs, can you update this with the DEV version of the fusion summary files?

Also, can you update the YAML file [here](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/.github/workflows/continuous_integration.yml) to add the ATRT subtyping module? Thanks!

File: analyses/molecular-subtyping-ATRT/ATRT_subtyping.R [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# create a list for ATRT subtypes
+ATRT_subtype_list <- c("ATRT, MYC", "ATRT, SHH", "ATRT, TYR")
+
+# for the samples, whose cns_methlation_subclass_score >= 0.8 and cns_methylation_subclass is one of the three types in ATRT_subtype_list, their molecular subtype are same as cns_methylation_subclass
+# for the samples without methylation sequencing, their molecular subtype are "ATRT, To be classified."
+# For the samples fit all the other situations, their molecular subtype are "ATRT, To be classified."
+atrt_subtype <- atrt_subtype %>% 
+  mutate(molecular_subtype = case_when(cns_methylation_subclass_score >= 0.8 & cns_methylation_subclass %in% ATRT_subtype_list ~ cns_methylation_subclass, 
+                                       is.na(Kids_First_Biospecimen_ID_meth) ~ "ATRT, To be classified.",
+                                       TRUE ~ "ATRT, To be classified.")) %>%
```
Comment (jharenza) [inline analyses/molecular-subtyping-ATRT/ATRT_subtyping.R]:
```suggestion
                                       is.na(Kids_First_Biospecimen_ID_meth) ~ "ATRT, To be classified",
                                       TRUE ~ "ATRT, To be classified")) %>%
```

Comment (jharenza) [reply analyses/molecular-subtyping-ATRT/ATRT_subtyping.R]: remove period

File: analyses/molecular-subtyping-ATRT/ATRT_subtyping.R [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  select(sample_id, Kids_First_Biospecimen_ID_meth, Kids_First_Biospecimen_ID_DNA, Kids_First_Biospecimen_ID_RNA, molecular_subtype) %>%
+  
+# write result
+  readr::write_tsv(file.path(results_dir, "ATRT_subtype_methylation.tsv"))
```
Comment (jharenza) [inline analyses/molecular-subtyping-ATRT/ATRT_subtyping.R]:
```suggestion
  readr::write_tsv(file.path(results_dir, "ATRT-molecular-subtypes.tsv"))
```
to be consistent with other module filenames

File: analyses/molecular-subtyping-ATRT/ATRT_subtyping.R [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Set path to result
+results_dir <-
+  file.path(root_dir, "analyses", "molecular-subtyping-ATRT", "results")
+
+
+# Read in histologies
+histo <- 
+  readr::read_tsv(file.path(root_dir, "data", "histologies.tsv"), guess_max = 10000)
```
Comment (jharenza) [inline analyses/molecular-subtyping-ATRT/ATRT_subtyping.R]:
```suggestion
histo <- 
  readr::read_tsv(file.path(root_dir, "data", "histologies-base.tsv"), guess_max = 10000)
```
use base in subtyping

**Comment by jharenza** [comment]
@zzgeng can you make the bash script in the ATRT subtyping module executable

File: analyses/molecular-subtyping-ATRT/Archive/run-molecular-subtyping-ATRT.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Chante Bethell for CCDL 2019
```
Comment (jharenza) [inline analyses/molecular-subtyping-ATRT/Archive/run-molecular-subtyping-ATRT.sh]:
```suggestion
# Chante Bethell for CCDL 2019, Zhuanzhuang Geng 2023
```

**Comment by jharenza** [comment]
Also, just to be consistent with all other modules, can you lowercase the A in Archive folder? Thanks!

File: analyses/molecular-subtyping-ATRT/Archive/run-molecular-subtyping-ATRT.sh (hunk 4 above)
Comment (zzgeng) [reply analyses/molecular-subtyping-ATRT/Archive/run-molecular-subtyping-ATRT.sh]: This is the bash script in archive folder. Should I make change here?

Comment (jharenza) [reply analyses/molecular-subtyping-ATRT/Archive/run-molecular-subtyping-ATRT.sh]: oh whoops - no you should delete this one fully. Sorry, didn't realize you had two in here!

**Review by jharenza — APPROVED** [review]
@zzgeng if this passes, we are good to go - updated the bash script to be executable and removed some unnecessary filter steps. I also noticed we were not collecting DGD samples in the final file, so added those to get the full cohort of ATRT here.

**Pull request merged**